airbyte-integrations/bases/debezium/src/main/java/io/airbyte/integrations/debezium/internals/postgres/PostgresDebeziumStateUtil.java

```diff
@@ -224,4 +233,39 @@ private OptionalLong extractLsn(final Set<Partition> partitions,
 
   }
 
+  public JsonNode constructInitialDebeziumState(final JdbcDatabase database, final String dbName) {
+    try {
+      return format(currentXLogLocation(database), currentTransactionId(database), dbName, Instant.now());
+    } catch (SQLException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @VisibleForTesting
+  public JsonNode format(final Long currentXLogLocation, final Long currentTransactionId, final String dbName, final Instant time) {
```

Comment:
Remind me again : why do we need current txn id in the DBZ/CDC state?

Reply:
Debezium expects it! Didnt deep dive into how it uses it